lib/gtag.js

```diff
@@ -0,0 +1,29 @@
+export const GA_TRACKING_ID = process.env.GA_TRACKING_ID;
+
+// https://developers.google.com/analytics/devguides/collection/gtagjs/pages
+export const pageview = (url) => {
+  window.gtag("config", GA_TRACKING_ID, {
+    page_path: url,
+  });
+};
+
+// https://developers.google.com/analytics/devguides/collection/gtagjs/events
+export const event = ({ action, category, label, value }) => {
+  window.gtag("event", action, {
+    event_category: category,
+    event_label: label,
+    value: value,
+  });
+};
+
+export function reportWebVitals({ id, name, label, value }) {
```

Comment:
Can make this typescript as well

```suggestion
export function reportWebVitals({ id, name, label, value } :  NextWebVitalsMetric) {
```

You will also have to import the type as follows

```
import { NextWebVitalsMetric } from 'next/app'
```